DOCUMENTATION.md

```diff
@@ -1,36 +1,41 @@
 # Documentation
+
 This is the documentation of the IPCLib, the layout of this document is the following:
 - ServerSocket
 - ClientSocket
 - Setting up a connection and sending data
 
 # ServerSocket
 
-A class that represents the serverside of a socket connection. It has functions to asynchronously connect and receive data from a client, and a synchronous function to send data to a client.
+A class that represents the serverside of a socket connection. It has functions to asynchronously connect and receive data from a client and a synchronous function to send data to a client.
+
+## Initialize
+
+Initializes and starts the server.
 
 ## ConnectAsync
 
 Spawns a child thread that waits for a connection, such that the main thread is not blocked.
 
 ## AwaitClientConnection
 
-The main thread awaits untill a client has connected to the server, this makes sure that the server cannot send data to a client that does not exist. It is best practise to call this function after ConnectAsync().
+The main thread awaits untill a client has connected to the server, this makes sure that the server cannot send data to a client that is not connected. If this function is called without ConnectAsync() the main thread will connect with a client. Otherwise the main thread will wait on the sub thread until it has connected.
 
 ## SendData
 
-Sends data to a connected client.
+Sends data to a connected client on the main thread. This data cannot be longer than IPC_BUFFER_BYTE_SIZE.
 
 ## ReceiveDataAsync
 
 Spawns a child thread that waits for data from a client, such that the main thread is not blocked.
 
 ## GetData
 
-Checks if data has been received in the child thread. If data was received it will return true, with the data in the buffers. If it did not it will return false.
+Checks if data has been received in the child thread. If data was received it will return true, with the data in the buffers. If it did not it will return false. 
 
 ## AwaitData
 
-Waits untill data is received from the client. Note that this function should ONLY be called after a ReceiveDataAsync() call, as otherwise the server will crash.
+Waits untill data is received from the client. If this function is called without ReceiveDataAsync() the main thread will wait until data is received on the socket. Otherwise it will wait until the sub thread returned data.
```

Comment:
By "called without ReceiveDataAsync()", do you mean "called before calling ReceiveDataAsync()"? The current phrasing indicates ReceiveDataAsync() would be passed as a parameter.